flink-runtime/src/main/java/org/apache/flink/runtime/topology/PipelinedRegion.java

```diff
@@ -0,0 +1,38 @@
+package org.apache.flink.runtime.topology;
+
+/**
+ * A pipelined region is a set of vertices connected via pipelined data
+ * exchanges.
+ *
+ * @param <VID> the type of the vertex ids
+ * @param <RID> the type of the result ids
+ * @param <V> the type of the vertices
+ * @param <R> the type of the result
+ */
+public interface PipelinedRegion<VID extends VertexID, RID extends ResultID,
+	V extends Vertex<VID, RID, V, R>, R extends Result<VID, RID, V, R>> {
+
+	/**
+	 * Returns vertices that are in this pipelined region.
+	 *
+	 * @return Iterable over all vertices in this pipelined region
+	 */
+	Iterable<V> getVertices();
+
+	/**
+	 * Returns the vertex with the specified vertex id.
+	 *
+	 * @param vertexId the vertex id used to look up the vertex
+	 * @return the vertex with the specified id
+	 * @throws IllegalArgumentException if there is no vertex in this pipelined
+	 *                                  region with the specified vertex id
+	 */
+	V getVertex(VID vertexId);
```

Comment:
I assume the FLIP is wrong here:

![image](https://user-images.githubusercontent.com/1681921/78556332-1c212c00-780f-11ea-8594-59662a00a735.png)

Reply:
You are right. I've just updated the FLIP.